pkg/clusteragent/admission/mutate/autoinstrumentation/config.go

```diff
@@ -373,6 +384,28 @@ func getOptionalBoolValue(datadogConfig config.Component, key string) *bool {
 	return value
 }
 
+func setStorageMedium(medium corev1.StorageMedium) {
```

Comment:
This certainly has a smell to it. The alternative though is to conduct a pretty large refactor to remove global volume variables and pass the medium through composition.